ocs_ci/utility/prometheus.py

```diff
@@ -414,27 +415,28 @@ def get(self, resource, payload=None):
         logger.debug(f"verify={self._cacert}")
         logger.debug(f"params={payload}")
 
-        response = requests.get(
-            self._endpoint + pattern,
-            headers=headers,
-            verify=self._cacert,
-            params=payload,
-        )
-        if "Application is not available" in response.text:
-            logger.warning(f"There was an error in response: {response.text}")
-            logger.warning("Refreshing connection")
-            self.refresh_connection()
-            if not config.ENV_DATA["platform"].lower() == "ibm_cloud":
-                logger.warning("Generating new certificate")
-                self.generate_cert()
-            logger.warning("Connection refreshed - trying the query again")
-            response = requests.get(
-                self._endpoint + pattern,
-                headers=headers,
-                verify=self._cacert,
-                params=payload,
-            )
-        return response
+        for response in TimeoutIterator(
+            timeout=300,
+            sleep=15,
+            func=requests.get,
+            func_kwargs={
+                "url": self._endpoint + pattern,
+                "headers": headers,
+                "verify": self._cacert,
+                "params": payload,
```

Comment:
Didn't you mention that you plan to tweak a timeout in the get call as well?

Reply:
No, I mentioned only that both TimeoutSampler and requests.get functions have timeout parameter and it can cause a conflict. After that you advised me to use TimeoutIterator class that should solve this conflict in parameters because parameters are passed as dictionary.